packages/plugin-testing-integration/src/extensions/bind.js

```diff
@@ -0,0 +1,17 @@
+export default (ns, oc) => {
+  const router = oc.get('$Router');
+  const nativeRoute = router.route.bind(router);
+
+  router.route = (path, ...args) => {
+    // We have to set correct url in jsdom for first application
+    // navigation to simulate browser behavior, where you
+    // already have correct url set in address bar.
+    if (!router._pageManager._managedPage.controller) {
```

Comment:
Better approach is take $PageManager from oc or the best use own boolean value.

Reply:
Added isFirstNavigation variable for better code readability and PageManager is now accessed from oc.